recipes-bsp/u-boot/files/0028-board-siemens-iot2050-implement-the-load-and-configu.patch

```diff
@@ -0,0 +1,342 @@
+From 52659c4bb91658ff6adcc39207a06612b3987e8f Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Fri, 7 Jan 2022 16:35:43 +0800
+Subject: [PATCH 28/28] board:siemens:iot2050:implement the load and
+ configuration for m2 variant
+
+For normal boot, load and configure m.2 device through boot.scr
+
+For secure boot enable, go through UEFI, fixup the compatible
+before loading the efibootguard in case we can load correct m.2 different
+variant device tree.
```

Comment:
How are these two cases reflected in the code? You do not check for a mode. Also, I see no reason why boot.scr should play a role in legacy boot. The state is in the U-Boot env in `m2_manual_config`, isn't it?

Reply:
boot script would use env fdtfile.  meaning it would go through the boot scr method to load the kernel dtb.
for secure mode, fix up the compatible to load the correct dtb.

Reply:
yes.  It just record that distro boot and uefiboot are different in two ways. So this patch covers this two ways